app/controllers/employment/AddEmploymentController.scala

```diff
@@ -0,0 +1,88 @@
+import config.{AppConfig, ErrorHandler}
+import controllers.predicates.{AuthorisedAction, InYearAction}
+import forms.YesNoForm
+import models.{IncomeTaxUserData, User}
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
+import services.EmploymentSessionService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import utils.{Clock, SessionHelper}
+import views.html.employment.AddEmploymentView
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class AddEmploymentController @Inject()(implicit val cc: MessagesControllerComponents,
+                                        authAction: AuthorisedAction,
+                                        inYearAction: InYearAction,
+                                        addEmploymentView: AddEmploymentView,
+                                        appConfig: AppConfig,
+                                        employmentSessionService: EmploymentSessionService,
+                                        errorHandler: ErrorHandler,
+                                        ec: ExecutionContext,
+                                        clock: Clock) extends FrontendController(cc) with I18nSupport with SessionHelper {
+
+
+  def show(taxYear: Int): Action[AnyContent] = authAction.async { implicit user =>
+    redirectOrRenderView(taxYear){
+      val form = if (SessionValues.EMPLOYMENT_ID.isEmpty) buildForm else buildForm.fill(value = true)
+      Future(Ok(addEmploymentView(form, taxYear)))
+    }
+  }
+
+  def submit(taxYear: Int): Action[AnyContent] = authAction.async { implicit user =>
+    redirectOrRenderView(taxYear){
+      submitForm(taxYear)
+    }
+  }
+
+  private def buildForm(implicit user: User[_]): Form[Boolean] = YesNoForm.yesNoForm(
+    missingInputError = "AddEmployment.error"
+  )
+
+
+  private def submitForm(taxYear:Int)(implicit user:User[_]): Future[Result] = {
+    buildForm.bindFromRequest().fold(
+      { formWithErrors =>
+        Future.successful(BadRequest(addEmploymentView(formWithErrors, taxYear)))
+      },
+      { radioValue =>
+        radioValue match {
+          case true => val id = UUID.randomUUID
+            Future.successful(Redirect(controllers.employment.routes.EmployerNameController.show(taxYear, id))
+              .addingToSession(SessionValues.EMPLOYMENT_ID -> id))
+          case _ => Future.successful(Redirect(appConfig.incomeTaxSubmissionOverviewUrl(taxYear)))
+        }
+      })
+  }
+
+  private def redirectOrRenderView(taxYear:Int)(f: => Future[Result])(implicit user:User[_]): Future[Result] = {
+    inYearAction.notInYear(taxYear) {
+      employmentSessionService.getPriorData(taxYear).flatMap {
+        case Right(IncomeTaxUserData(Some(_))) => Future(Redirect(controllers.employment.routes.EmploymentSummaryController.show(taxYear)))
+        case Right(IncomeTaxUserData(None)) => f
```

Comment:
I'd maybe rename "f" here. It kind of makes sense, but it's hard to read and looks weird